
Status: in BTS
Note: also needs libtool, config.{guess,sub} and config.rpath update

diff -ur libgd2-2.0.28.old/debian/rules libgd2-2.0.28/debian/rules
--- libgd2-2.0.28.old/debian/rules	2004-08-26 21:31:47.000000000 +0200
+++ libgd2-2.0.28/debian/rules	2004-08-27 00:31:05.000000000 +0200
@@ -22,11 +22,14 @@
 TOOLS =	pngtogd pngtogd2 gdtopng gd2topng gd2copypal gdparttopng webpng
 TESTS = gddemo gdtest
 
+export PTHREAD_LIBS = -lpthread
+
 configure: configure-stamp
 configure-stamp:
 	dh_testdir
 	chmod -R a=r,a+X,u+w *
 	chmod a+x debian/rules configure
+	cp -f /usr/share/misc/config.* config/
 	-mkdir $(BUILD-ARCH-TARGETS)
 	cd $(CURDIR)/build-libgd2-noxpm && ../configure --prefix=/usr --without-x --disable-rpath
 	cd $(CURDIR)/build-libgd2-xpm && ../configure --prefix=/usr --with-x --disable-rpath
--- libgd2-2.0.28.old/configure.ac	2004-07-21 20:55:32.000000000 +0200
+++ libgd2-2.0.28/configure.ac	2004-08-27 00:31:05.000000000 +0200
@@ -93,12 +93,15 @@
         AC_MSG_CHECKING([for pthread_join in LIBS=$PTHREAD_LIBS with CFLAGS=$PTHREAD_CFLAGS])
         AC_TRY_LINK_FUNC(pthread_join, acx_pthread_ok=yes)
         AC_MSG_RESULT($acx_pthread_ok)
+        LIBS="$save_LIBS"
+        CFLAGS="$save_CFLAGS"
         if test x"$acx_pthread_ok" = xno; then
                 PTHREAD_LIBS=""
                 PTHREAD_CFLAGS=""
+	else
+	        LIBS="$PTHREAD_LIBS $LIBS"
+	        CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
         fi
-        LIBS="$save_LIBS"
-        CFLAGS="$save_CFLAGS"
 fi
 
 # We must check for the threads library under a number of different
